test/authority_test.go

```diff
@@ -33,44 +34,48 @@ import (
 	testpb "google.golang.org/grpc/test/grpc_testing"
 )
 
+func authorityChecker(expectedAuthority *string) func(context.Context, *testpb.Empty) (*testpb.Empty, error) {
```

Comment:
I'm not sure about passing a pointer here without synchronization.  It may fail the race test.

I'd do this instead:

```go
func authorityChecker(ctx context.Context, expectedAuthority string) (*testpb.Empty, error) {
  // Same body as closure
}

...

ss := &stubServer{
	emptyCall: func(ctx context.Context, _ *testpb.Empty) (*testpb.Empty, error) {
		// Simple case:
		return authorityChecker(ctx, wantAuthority)

		// Complex case:
		wantAuthorityMu.Lock()
		defer wantAuthorityMu.Unlock()
		return authorityChecker(ctx, wantAuthority)
	}
}
```

Or use a channel to pass the expected authority to the handler, but that's probably more trouble than it's worth.

Reply:
Made this change.